packages/tabs/src/utils/getTestUtils/getTestUtils.ts

```diff
@@ -56,18 +55,50 @@ export const getTestUtils = (
   };
 
   /**
-   * Queries the `element` for the selected panel. Returns null if selected panel is not found.
+   * Queries the `element` for the tab panels element. Will throw if no element is found.
+   * Then, finds and returns all elements with role=tabpanel. Will throw if no tab panels are found.
    */
-  const getSelectedPanel = () => {
-    return queryBySelector<HTMLElement>(
+  const getAllTabPanelsInDOM = () => {
+    const tabPanels = queryBySelector<HTMLElement>(
       element,
-      '[role="tabpanel"]:not(:empty)',
+      `[data-lgid=${LGIDS_TABS.tabPanels}]`,
+    );
+
+    if (!tabPanels) {
+      throw new Error('Unable to find tab panels container');
+    }
+
+    const allTabPanels =
+      tabPanels.querySelectorAll<HTMLElement>('[role="tabpanel"]');
+
+    if (allTabPanels.length === 0) {
+      throw new Error(
+        'Unable to find any tabpanel elements in tab panels container',
+      );
+    }
+
+    return Array.from(allTabPanels);
+  };
+
+  /**
+   * Gets all tab panels and filters for the displayed tab panel. Returns null if selected panel is not found.
+   */
+  const getSelectedPanel = () => {
+    const allTabPanels = getAllTabPanelsInDOM();
+
+    const visibleTabPanel = allTabPanels.find(
+      tabPanel => getComputedStyle(tabPanel).display !== 'none',
```

Comment:
Should a selected tab panel have `aria-selected`?

Reply:
looks like im wrong. `aria-selected` is for `tab`, not `tabpanel`.

Reply:
Another alternative could be to use the `hidden` attribute like [mdn docs](https://developer.mozilla.org/en-US/play), and you can search for that attribute?

Reply:
from what I could find, this was the recommended approach over `hidden` attribute

source: https://stackoverflow.com/questions/6708247/what-is-the-difference-between-the-hidden-attribute-html5-and-the-displaynone

we also have to use an approach that satisfies search engine crawling. It's still not 100% bulletproof, will depend on how it fares in the docs team implementation